zap.zsh

```diff
@@ -89,70 +89,80 @@ delete() {
-        sed -i'.backup' "/$usr\/$plg/s/^/#/g" $ZAP_ZSHRC
+        sed -i'.backup' "/$usr\/$plg/s/^/#\ /g" $ZAP_ZSHRC
         rm -rf $ZAP_PLUGIN_DIR/$plg && echo "Deleted $plg" || echo "Failed to Delete $plg"
     done
 }
 
-pause() {
+_zap_deactivate() {
     plugins=$(awk 'BEGIN { FS = "[ plug]" } { print }' $ZAP_ZSHRC | grep -E '^plug "' | awk 'BEGIN { FS = "[ \"]" } { print " " int((NR)) echo "  🔌 " $3 }')
     echo "$plugins \n"
     echo -n "🔌 Plugin Number | (a) All Plugins: "
     read plugin
     echo ""
     if [[ $plugin == "a" ]]; then
-        sed -i'.backup' '/^plug/s/^/#/g' $ZAP_ZSHRC
+        sed -i'.backup' '/^plug/s/^/#\ /g' $ZAP_ZSHRC
     else
         for plug in $plugins; do
             usr=$(echo $plug | grep $plugin | awk 'BEGIN { FS = "[ /]" } { print $5 }')
             plg=$(echo $plug | grep $plugin | awk 'BEGIN { FS = "[ /]" } { print $6 }')
-            sed -i'.backup' "/$usr\/$plg/s/^/#/g" $ZAP_ZSHRC
+            sed -i'.backup' "/$usr\/$plg/s/^/#\ /g" $ZAP_ZSHRC
         done
     fi
 }
 
-unpause() {
+_zap_activate() {
     plugins=$(awk 'BEGIN { FS = "[ plug]" } { print }' $ZAP_ZSHRC | grep -E '^#plug "' | awk 'BEGIN { FS = "[ \"]" } { print " " int((NR)) echo "  🔌 " $3 }')
     echo "$plugins \n"
     echo -n "🔌 Plugin Number | (a) Plug All: "
     read plugin
     echo ""
     if [[ $plugin == "a" ]]; then
-        sed -i'.backup' '/^#plug/s/^#//g' $ZAP_ZSHRC
+        sed -i'.backup' '/^#\ plug/s/^#\ //g' $ZAP_ZSHRC
     else
         for plug in $plugins; do
             usr=$(echo $plug | grep $plugin | awk 'BEGIN { FS = "[ /]" } { print $5 }')
             plg=$(echo $plug | grep $plugin | awk 'BEGIN { FS = "[ /]" } { print $6 }')
-            sed -i'.backup' "/$usr\/$plg/s/^#//g" $ZAP_ZSHRC
+            sed -i'.backup' "/$usr\/$plg/s/^#\ //g" $ZAP_ZSHRC
         done
     fi
 }
 
-help() {
+_zap_help() {
     cat "$ZAP_DIR/doc.txt"
 }
 
-version() {
+_zap_version() {
     ref=$ZAP_DIR/.git/packed-refs
     tag=$(awk 'BEGIN { FS = "[ /]" } { print $3, $4 }' $ref | grep tags)
     ver=$(echo $tag | cut -d " " -f 2)
     echo "\n ⚡Zap Version v$ver \n"
 }
 
+typeset -A opts
+opts=(
+    -a "_zap_activate"
+    --activate "_zap_activate"
+    -d "_zap_deactivate"
+    --deactivate "_zap_deactivate"
+    -h "_zap_help"
+    --help "_zap_help"
+    -r "_zap_remove"
+    --remove "_zap_remove"
+    -u "_zap_update"
+    --update "_zap_update"
+    -v "_zap_version"
+    --version "_zap_version"
+)
+
 zap() {
-    local command="$1"
-    if [[ $command == "-v" ]] || [[ $command == "--version" ]]; then
-        version
-        return
+    emulate -L zsh
+    if [[ -z "$opts[$1]" ]]; then
+        _zap_help
+        return 1
     else
-        if [[ $command == "-h" ]] || [[ $command == "--help" ]]; then
-            help
-            return
-        else
-            $command
-            return
-        fi
-        echo "$command: command not found"
+        opt="${opts[$1]}"
```

Comment:
I think we can skip the else case since we are returning in the if statement above. Big fan of early return and skip else cases if possible :)